ublox/src/ubx_packets/packets.rs

```diff
@@ -1970,15 +1973,55 @@ struct RxmRtcm {
 }
 
 #[ubx_packet_recv]
-#[ubx(class = 0x10, id = 0x02, fixed_payload_len = 16)]
+#[ubx(class = 0x10, id = 0x02, max_payload_len = 1240)]
 struct EsfMeas {
     time_tag: u32,
+    #[ubx(map_type = EsfMeasInfo, from = EsfMeasInfo::new)]
+    info: [u8; 0],
+}
+
+pub struct EsfMeasInfo<'a> {
     flags: u16,
     id: u16,
-    /// flags >> 11 & 0x1F
-    data: u32,
-    /// flags & 0x8
-    calib_tag: u32,
+    data: Box<dyn Iterator<Item = &'a u32>>,
+    calib_tag: Option<u32>,
+}
+
+impl<'a> EsfMeasInfo<'a> {
+    fn new(bytes: &'a [u8]) -> Self {
+        let mut chunks = bytes.chunks_exact(2);
+        let flags = u16::from_le_bytes(chunks.next().unwrap().try_into().unwrap());
+        let id = u16::from_le_bytes(chunks.next().unwrap().try_into().unwrap());
+
+        let calib_tag = if flags & 0x8 != 0 {
+            let first_last: [u8; 2] = chunks.next_back().unwrap().try_into().unwrap();
+            let second_last: [u8; 2] = chunks.next_back().unwrap().try_into().unwrap();
+            Some(u32::from_le_bytes(
+                [first_last, second_last].concat().try_into().unwrap(),
+            ))
+        } else {
+            None
+        };
+
+        let data = Box::new(
+            chunks
+                .tuples::<(&[u8], &[u8])>()
+                .map(|(a, b)| u32::from_le_bytes([a, b].concat().try_into().unwrap())),
+        ) as Box<dyn Iterator<Item = &'a u32>>;
```

Comment:
Managed to rewrite into something like this...
But the lifetimes are a bit wonky so it doesn't compile

Not sure if something like this would work, think `.concat()` turns it into a `Vec` so not sure with the `no std / no alloc` thing

Reply:
So `Box` will actually have the same issues as `Vec`, with `no_std` and no `alloc` you can't use any type that puts data on the heap which includes `Box`

Reply:
that is so annoying... guess we'll have to settle with another iterator